### PR TITLE
update for use with file base locators

### LINKED ISSUE #33: locator pallet
Is this pallet is working?

image

~~Should the [sddrafts folder](https://github.com/agrc/warehouse/blob/master/mapserv/LocatorsPallet.py#L48) be published to github?~~

It appears to be a forgotten secret. I updated the readme and added the missing items.

If I am reading this pallet correctly, the locators need to be in the `locators.gdb` prior to a run. This is problematic for a few reasons. We should be able to purge the staging data and have things rebuild properly. 

A corrupt locator has no way to recover without intervention. 

> Failed to execute. Parameters are not valid. ERROR 000732: Input Address Locator: Dataset staging\locators.gdb\AddressPoints_AddressSystem does not exist or is not supported Failed to execute (RebuildAddressLocator).

Keeping a gdb of valid locators with the project in github could give us the opportunity to generate the locators.gdb by `arcpy.copy_management`. We could update the data, (delete if exists) copy the valid locator into staging, rebuild, and finally publish. 

Thoughts?

### PR DESCRIPTION
**Pull request by steveoh**
requires https://github.com/agrc/forklift/pull/200

## Usage Instructions

1. Update the `secrets.py` based on the template. _this pallet uses the configuration you run forklift with_
1. from a python repl
`from LocatorsPallet import LocatorsPallet as L; l = L(); l.build('Dev'); l.create_locators()`
1. If you want the lightswitch code to work, publish the locators to `Geolocators`

closes #33

### CODE REVIEW

File: mapserv/LocatorsPallet.py [hunk 1]
```diff
@@ -1,83 +1,330 @@
...
+from os import mkdir
+from os.path import join
+from os.path import split
+from shutil import copyfile
+from shutil import rmtree
+from time import clock
 
-from agrc.agrc import ags
 
+class LocatorPallet(Pallet):
```
Comment (stdavis) [inline mapserv/LocatorsPallet.py]: This should match the filename (`LocatorsPallet`).

File: mapserv/LocatorsPallet.py [hunk 2]
```diff
@@ -1,83 +1,330 @@
...
+
+    def copy_locator_to(self, file_path, locator, to_folder):
+        location = join(file_path, locator)
+        self.log.debug('copying %s to %s', location, to_folder) #: iterator glob for .lox .loc .loc.xml
+        for filename in iglob(location + '.lo*'):
+            base_folder, locator_with_extension = split(filename)
+
+            try:
+                mkdir(to_folder)
+            except FileExistsError:
```
Comment (stdavis) [inline mapserv/LocatorsPallet.py]: This is undefined for me

File: mapserv/LocatorsPallet.py [hunk 3]
```diff
@@ -1,83 +1,330 @@
...
-        serviceType = 'GeocodeServer'
-        self.log.info('deleting existing service')
+        self.log.debug('rebuilding %s', locator)
+
+        #: rebuild in temp location
+        arcpy.geocoding.RebuildAddressLocator(locator)
+
+    def copy_locator_to(self, file_path, locator, to_folder):
+        location = join(file_path, locator)
+        self.log.debug('copying %s to %s', location, to_folder) #: iterator glob for .lox .loc .loc.xml
```
Comment (stdavis) [inline mapserv/LocatorsPallet.py]: pep8 doesn't like this. At least two spaces before inline comment.

File: mapserv/LocatorsPallet.py (hunk 2 above)
Comment (stdavis) [inline mapserv/LocatorsPallet.py]: Oh wait. I can see it's built in at py3. I need to figure out how to make my pep8 use py3...

Comment (steveoh) [inline mapserv/LocatorsPallet.py]: did you fill out your secrets for the configuration you are running forklift for?

Comment (stdavis) [inline mapserv/LocatorsPallet.py]: Yes. But not sure how this relates to linting. I was able to get flake8 to work with python three buy running `pip3 install flake8`.

Comment (steveoh) [inline mapserv/LocatorsPallet.py]: Conda has the flake8 module

Comment (stdavis) [inline mapserv/LocatorsPallet.py]: Ya, I was seeing it in Atom.

**Comment by stdavis** [comment]
My tests never finished with this but it looks good otherwise to me.

**Comment by steveoh** [comment]
What happened with your tests @stdavis

**Pull request merged**